packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -455,6 +528,73 @@ class MemoryTimeline {
 
   ValueNotifier<bool> get pausedNotifier => _pausedNotifier;
 
+  dart_ui.Image _img;
+
+  set image(dart_ui.Image img) {
+    _img = img;
+  }
+
+  /// Common utility function to handle loading of the data into the
+  /// chart for either offline or live Feed.
+  List<Map> _processData(List<HeapSample> data, int startingDataIndex) {
+    final result = <Map<String, Entry>>[];
+
+    for (var dataIndex = startingDataIndex;
+        dataIndex < data.length;
+        dataIndex++) {
+      final sample = data[dataIndex];
+      final timestamp = sample.timestamp.toDouble();
+
+      final capacity = sample.capacity.toDouble();
+      final used = sample.used.toDouble();
+      final external = sample.external.toDouble();
+
+      final extEntry = Entry(
+        x: timestamp,
+        y: external,
+        icon: _img,
+      );
+      final usedEntry = Entry(
+        x: timestamp,
+        y: used + external,
+        icon: _img,
+      );
+      final capacityEntry = Entry(
+        x: timestamp,
+        y: capacity,
+        icon: _img,
+      );
+
+      result.add({
+        capcityValueKey: capacityEntry,
+        usedValueKey: usedEntry,
+        externalValueKey: extEntry,
+      });
+    }
+
+    return result;
+  }
+
+  /// Fetch all the data in the loaded from a memory log (JSON file in /tmp).
+  List<Map> processMemoryLogFileData() {
+    assert(controller.offline);
+    assert(offlineData.isNotEmpty);
+    return _processData(offlineData, 0);
+  }
+
+  List<Map> processLiveData([bool reloadAllData = false]) {
+    assert(!controller.offline);
+    assert(liveData.isNotEmpty);
+
+    final List<HeapSample> liveFeed = liveData;
```

Comment:
Is this intended to make a copy (`List.from(liveData)`)? As is it isn't doing anything but assigning another reference to the object

Reply:
No, just old code where I was accessing either liveData of offlienData now its only liveData.  Removed local.